rdr_service/services/consent/files.py

```diff
@@ -476,16 +502,176 @@ def is_agreement_selected(self):
             return False
 
 
+class CeFileWrapper:
+    def __init__(self, pdf):
+        self.pdf = pdf
+
+    def get_signature_on_file(self):
+        signature_page = self._get_last_page()
+        signature_footer_location = self._get_location_of_string(signature_page, "Participant's Name (printed)")
+
+        if signature_footer_location:
+            signature_string_list = self._text_in_bounds(
+                search_rect=self._rect_shifted_up_from_footer(signature_footer_location),
+                element=signature_page
+            )
+            if signature_string_list:
+                return signature_string_list[0]
+
+        return None
+
+    def get_date_signed_str(self):
+        signature_page = self._get_last_page()
+        signature_footer_location = self._get_location_of_string(signature_page, 'Date')
+
+        if signature_footer_location:
+            date_string_list = self._text_in_bounds(
+                search_rect=self._rect_shifted_up_from_footer(signature_footer_location),
+                element=signature_page
+            )
+            if date_string_list:
+                return date_string_list[0]
+
+        return None
+
+    def _get_last_page(self):
+        last_page_index = len(self.pdf.pages) - 1
+        return self.pdf.pages[last_page_index]
+
+    @classmethod
+    def _rect_shifted_up_from_footer(cls, footer_rect: Rect) -> Rect:
+        return Rect.from_edges(
```

Comment:
Is your process for determining these values documented somewhere?

Reply:
No, this was just from working with the files and finding the relative positioning from the labels. How do you think this should be documented?

Reply:
I think it would be helpful for maintenance if there was a (informal) tech doc of some kind that captured how you figured out the offsets, or any other "trial and error"(?)-type processes on how you worked with the `pdf-miner` library In case someone else is trying to do the same kind of validation development for a different consent down the line and is leveraging from existing code.  (I know there's limited bandwidth for going the extra mile for more documentation right now, so just something to keep in mind)

Reply:
Thanks, I'll keep it in mind.